echo-notifications/src/main/groovy/com/netflix/spinnaker/echo/microsoftteams/MicrosoftTeamsService.java

```diff
@@ -16,18 +16,64 @@
 
 package com.netflix.spinnaker.echo.microsoftteams;
 
+import com.netflix.spinnaker.retrofit.Slf4jRetrofitLogger;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 import lombok.extern.slf4j.Slf4j;
+import retrofit.RestAdapter;
+import retrofit.client.Client;
 import retrofit.client.Response;
+import retrofit.converter.JacksonConverter;
 
 @Slf4j
 public class MicrosoftTeamsService {
-  private final MicrosoftTeamsClient microsoftTeamsClient;
+  private static final String BASE_URL_REGEX_PATTERN = "^(http|https)://.+?/";
 
-  public MicrosoftTeamsService(MicrosoftTeamsClient microsoftTeamsClient) {
-    this.microsoftTeamsClient = microsoftTeamsClient;
+  private Client retrofitClient;
+  private RestAdapter.LogLevel retrofitLogLevel;
+
+  public MicrosoftTeamsService(Client retrofitClient, RestAdapter.LogLevel retrofitLogLevel) {
+    this.retrofitClient = retrofitClient;
+    this.retrofitLogLevel = retrofitLogLevel;
   }
 
   public Response sendMessage(String webhookUrl, MicrosoftTeamsMessage message) {
-    return microsoftTeamsClient.sendMessage(webhookUrl, message);
+    MicrosoftTeamsClient microsoftTeamsClient =
+        new RestAdapter.Builder()
+            .setConverter(new JacksonConverter())
+            .setClient(retrofitClient)
+            .setEndpoint(getEndpointUrl(webhookUrl))
+            .setLogLevel(retrofitLogLevel)
+            .setLog(new Slf4jRetrofitLogger(MicrosoftTeamsClient.class))
+            .build()
+            .create(MicrosoftTeamsClient.class);
+
+    return microsoftTeamsClient.sendMessage(getRelativePath(webhookUrl), message);
+  }
+
+  private String getEndpointUrl(String webhookUrl) {
+    String baseUrl = "";
+    Pattern pattern = Pattern.compile(BASE_URL_REGEX_PATTERN, Pattern.CASE_INSENSITIVE);
```

Comment:
can we factor this out and compile the pattern only once during the initialization of MicrosoftTeamsService?

Reply:
Looking back on this, I clearly was overthinking this and have no idea why I chose to do regex and not just use `java.net.URL`. 

I removed the regex use `java.net.URL` to parse the URL segments instead.